tests/test_utilities.py

```diff
@@ -156,6 +156,132 @@ def test_check_all_contiguous():
 
+def test_interleaveImages():
+    import time
+    t1 = time.time()
+    # 1) Interleave arrays as Images
+    x_size, y_size = 16,10
+    n1, n2 = 2, 5
+    im_list = []
+    for i in xrange(n1):
+        for j in xrange(n2):
+            x = np.arange(-x_size+2.*i/n1,x_size+2.*i/n1,2)
+            y = np.arange(-y_size+2.*j/n2,y_size+2.*j/n2,2)
+            X,Y = np.meshgrid(y,x)
+            im = galsim.Image(100.*X+Y)
+            im_list.append(im)
+
+    im1 = galsim.utilities.interleaveImages(im_list,N=(n2,n1),suppress_warnings=True)
+    x = np.arange(-x_size,x_size,2.0/n1)
+    y = np.arange(-y_size,y_size,2.0/n2)
+    X,Y = np.meshgrid(y,x)
+    im2 = galsim.Image(100.*X+Y)
+    np.testing.assert_array_almost_equal(im1.array,im2.array,decimal=11,\
+        err_msg="Interleave failed for dummy image")
+    assert im1.scale == None
+
+    # 2a) With galsim Gaussian
+    g = galsim.Gaussian(sigma=3.7,flux=1000.)
+    gal = galsim.Convolve([g,galsim.Pixel(1.0)])
+    im_list = []
+    n = 2
+    for j in xrange(n):
+        for i in xrange(n):
+            im = galsim.Image(16*n,16*n)
+            offset = galsim.PositionD(-(i+0.5)/n+0.5,-(j+0.5)/n+0.5)
+            gal.drawImage(image=im,scale=1.0,method='no_pixel',offset=offset)
+            #gal.drawImage(image=im,scale=1.0,method='no_pixel',offset=galsim.PositionD(1.*i/n,1.*j/n))
+            im_list.append(im)
+
+    # Input to N as an int
+    img = galsim.utilities.interleaveImages(im_list,n)
+    im = galsim.Image(16*n*n,16*n*n)
+    g = galsim.Gaussian(sigma=3.7,flux=1000.*n*n)
+    gal = galsim.Convolve([g,galsim.Pixel(1.0)])
+    gal.drawImage(image=im,method='no_pixel',offset=galsim.PositionD(0.0,0.0),scale=1.0/n)
+    np.testing.assert_array_equal(img.array,im.array,\
+        err_msg="Interleaved Gaussian images do not match")
+    assert im.scale == img.scale
+
+    # 2b) With im_list and offsets permuted
+    offset_list = []
+    # An elegant way of generating the default offsets
+    DX = np.arange(0.0,-1.0,-1.0/n)
+    DX -= DX.mean()
+    DY = DX
+    for dy in DY:
+        for dx in DX:
+            offset = galsim.PositionD(dx,dy)
+            offset_list.append(offset)
+
+    np.random.seed(42) # for generating the same random permutation everytime
+    rand_idx = np.random.permutation(len(offset_list))
+    im_list_randperm = [im_list[idx] for idx in rand_idx]
+    offset_list_randperm = [offset_list[idx] for idx in rand_idx]
+    # Input to N as a tuple
+    img_randperm = galsim.utilities.interleaveImages(im_list_randperm,(n,n),offsets=offset_list_randperm)
+
+    np.testing.assert_array_equal(img_randperm.array,img.array,\
+        err_msg="Interleaved images do not match when 'offsets' is supplied")
+    assert img_randperm.scale == img.scale
+
+    # 3a) Increase resolution along one direction - square to rectangular images
+    n = 2
+    g = galsim.Gaussian(sigma=3.7,flux=100.)
+    g1 = g.shear(g=1.*(n**2-1)/(n**2+1),beta=0.0*galsim.radians)
```

Comment:
Thanks! I had to justify why interleaving only along one direction is even a thing one would want to do.